src/SvgStream.h

```diff
@@ -122,7 +135,17 @@ class SvgStreamString : public SvgStream {
     if(!svgstr.empty()) {
       svgstr.append("</svg>");
     }
-    env_["svg_string"] = svgstr;
+    if (env_.exists("svg_string")) {
+      Rcpp::CharacterVector str = env_["svg_string"];
+      str.push_back(svgstr);
+      env_["svg_string"] = str;
+    } else {
+      env_["svg_string"] = svgstr;
+    }
+
+    // clear the stream
+    stream_.str(std::string());
+    stream_.clear();
```

Comment:
does this mean that `svgstring` always overwrite on new page? If so, I think it is wrong (and we may have misunderstood each other).

I think it is fine for `svgstring()` to always return a character vector with an element for each page. `svglite()` should overwrite the output on new page if, and only if, the filename doesn't support multiple pages. If that is the current behaviour then all is good

Reply:
> I think it is fine for svgstring() to always return a character vector with an element for each page. svglite() should overwrite the output on new page if, and only if, the filename doesn't support multiple pages. If that is the current behaviour then all is good

That's the current behavior.

In R:
```r
# example from unit test
df <- data.frame(x = rnorm(20), y = rnorm(20))
plot_one <- function() plot(df$x, df$y)
plot_two <- function() plot(df$x, df$y + 10)

# string with 2 plots
s_multiple <- svgstring()
plot_one()
plot_two()
dev.off()

svg_string <- s_multiple()
length(svg_string) # returns 2
svg_string[1] != svg_string[2] # returns TRUE
```
In general, `s_multiple()` returns a character vector where each element is a plot. 

In C++, the current string in the stream is pushed to a `CharacterVector`, and afterward the stream is emptied.

Reply:
Great!